src/include/distributed/metadata_sync.h

```diff
@@ -73,11 +73,10 @@ extern Oid GetAttributeTypeOid(Oid relationId, AttrNumber attnum);
 	"VALUES (" UINT64_FORMAT ", %d, " UINT64_FORMAT \
 	", %d, " UINT64_FORMAT \
 	") " \
-	"ON CONFLICT (placementid) DO UPDATE SET " \
-	"shardid = EXCLUDED.shardid, " \
+	"ON CONFLICT (shardid, groupid) DO UPDATE SET " \
 	"shardstate = EXCLUDED.shardstate, " \
 	"shardlength = EXCLUDED.shardlength, " \
-	"groupid = EXCLUDED.groupid"
+	"placementid = EXCLUDED.placementid"
```

Comment:
what triggered this change?

Reply:
I cannot remember the exact reason right now, one of the tests was failing. Let me get the tests passing state again, and then I'll find the failing test to remember the reason

Reply:
I think the reason is that we do not sync the metadata changes to a disabled node. Meaning that the placements that are deleted from the coordinator is not reflected on the disabled node.

When you re-activate the disabled node, we re-sync the metadata for reference tables at first via `ReplicateAllReferenceTablesToNode`, which is before we truncate the metadata in `SyncMetadata`. 

And, it fails as follows where the shardid belongs to a reference table:
```SQL
INSERT INTO pg_dist_placement (shardid, shardstate, shardlength, groupid, placementid) VALUES (101502, 1, 0, 2, 15) ON CONFLICT (placementid) DO UPDATE SET shardid = EXCLUDED.shardid, shardstate = EXCLUDED.shardstate, shardlength = EXCLUDED.shardlength, groupid = EXCLUDED.groupid;
ERROR:  duplicate key value violates unique constraint "placement_shardid_groupid_unique_index"
DETAIL:  Key (shardid, groupid)=(101502, 2) already exists.
```

And, we are generating a new `PlacementId` for the inactive node. So, the group/shardId will be the same, but the placementId would change.

Reply:
And, as I suspected, the UPSERT was useless on master, changing it to INSERT still passes all the tests: https://github.com/citusdata/citus/compare/fix_upser?expand=1